src/ol/source/ogcTileUtil.js

```diff
@@ -0,0 +1,376 @@
+/**
+ * @module ol/source/ogcTileUtil
+ */
+
+import TileGrid from '../tilegrid/TileGrid.js';
+import {assign} from '../obj.js';
+import {getJSON, resolveUrl} from '../net.js';
+import {get as getProjection} from '../proj.js';
+import {getIntersection as intersectExtents} from '../extent.js';
+
+/**
+ * See https://ogcapi.ogc.org/tiles/.
+ */
+
+/**
+ * @enum {string}
+ */
+const TileType = {
+  MAP: 'map',
+  VECTOR: 'vector',
+};
```

Comment:
I think you mean `{'map' | 'vector'}` (instead of the intersection).

This looks like a nice alternative where string values are used only once (or maybe only a couple times).  If we were to repeat the `'map'` and `'vector'` string literals many times throughout the library, I think I'd prefer to use the `TileType.MAP` form.

We'd have to run some tests to confirm, but I imagine the (pre-compression) build size would be smaller after a few repeat occurrences.  Maybe it would be interesting to compare the two approaches with something like the `ol/source/State.js` enum.